### ISSUE TITLE
I have one question, which training script can achieve the best performance?

### ISSUE THREAD

**Issue by hszhoushen** [ChCh1999/RTPB#5]
When I was using DT_base_cb to train on the predcls and sgcls tasks, it seems a distance from the reported results on the validation set. e.g., 

Or is the reason that I don't fine-tune the detector leading to worse performance on these tasks. If so, could you please provide the fine-tuned object detectors for us for convenience?

====================================================================================================
Detection evaluation mAp=0.9999
====================================================================================================
SGG eval:     R @ 20: 0.5396;     R @ 50: 0.6005;     R @ 100: 0.6207;  for mode=predcls, type=Recall(Main).
SGG eval:  ng-R @ 20: 0.6249;  ng-R @ 50: 0.7595;  ng-R @ 100: 0.8372;  for mode=predcls, type=No Graph Constraint Recall(Main).
SGG eval:    zR @ 20: 0.0000;    zR @ 50: 0.0000;    zR @ 100: 0.0000;  for mode=predcls, type=Zero Shot Recall.
SGG eval: ng-zR @ 20: 0.0000; ng-zR @ 50: 0.0089; ng-zR @ 100: 0.0356;  for mode=predcls, type=No Graph Constraint Zero Shot Recall.
SGG eval:    mR @ 20: 0.1914;    mR @ 50: 0.2424;    mR @ 100: 0.2596;  for mode=predcls, type=Mean Recall.

====================================================================================================
Detection evaluation mAp=0.5889
====================================================================================================
SGG eval:     R @ 20: 0.3296;     R @ 50: 0.3668;     R @ 100: 0.3788;  for mode=sgcls, type=Recall(Main).
SGG eval:  ng-R @ 20: 0.3759;  ng-R @ 50: 0.4560;  ng-R @ 100: 0.5012;  for mode=sgcls, type=No Graph Constraint Recall(Main).
SGG eval:    zR @ 20: 0.0044;    zR @ 50: 0.0089;    zR @ 100: 0.0089;  for mode=sgcls, type=Zero Shot Recall.
SGG eval: ng-zR @ 20: 0.0022; ng-zR @ 50: 0.0244; ng-zR @ 100: 0.0511;  for mode=sgcls, type=No Graph Constraint Zero Shot Recall.
SGG eval:    mR @ 20: 0.1139;    mR @ 50: 0.1338;    mR @ 100: 0.1418;  for mode=sgcls, type=Mean Recall.

[DT_base.sh]
[DT_base_cb.sh]
[DT_base_clsbal_reweight.sh]
[DT_base_eb.sh]
[DT_base_focal.sh]
[DT_base_ldam.sh]
[DT_base_pb.sh]
[DT_base_reweight.sh]
[DT_base_vb.sh]
[motif_base.sh]
[motif_base_cb.sh]
[vctree_base.sh]
[vctree_base_cb.sh]

**Comment by ChCh1999** [comment]
Thanks for your interest in our work. I'm not sure what happened in your experiment. 
I've uploaded the [detector's fine-tuned weight](https://github.com/ChCh1999/RTPB/blob/1f63b708118b52efe58806ca05291d35208af7a6/Models.md). 

You can use any script to reproduce the corresponding experiment, e.g. `DT_base_cb.sh` for `DTrans+RTPB(CB)`. For more detail, please refer to our [paper](https://arxiv.org/abs/2201.06794).

Here is my experiment result for predcls and sgcls with `DT_base_cb.sh`. Hope it help.
[PredCls](https://github.com/ChCh1999/RTPB/files/9129107/predcls_dt_cb.txt): 
```
====================================================================================================
Detection evaluation mAp=0.9995
====================================================================================================
SGG eval:     R @ 20: 0.3908;     R @ 50: 0.4562;     R @ 100: 0.4752;  for mode=predcls, type=Recall(Main).
SGG eval:  ng-R @ 20: 0.5076;  ng-R @ 50: 0.7003;  ng-R @ 100: 0.8071;  for mode=predcls, type=No Graph Constraint Recall(Main).
SGG eval:    zR @ 20: 0.0057;    zR @ 50: 0.0106;    zR @ 100: 0.0149;  for mode=predcls, type=Zero Shot Recall.
SGG eval: ng-zR @ 20: 0.0081; ng-zR @ 50: 0.0360; ng-zR @ 100: 0.0785;  for mode=predcls, type=No Graph Constraint Zero Shot Recall.
SGG eval:    mR @ 20: 0.3028;    mR @ 50: 0.3618;    mR @ 100: 0.3808;  for mode=predcls, type=Mean Recall.
```
[SGCls](https://github.com/ChCh1999/RTPB/files/8548981/sgcls_dt_cb.txt): 
```
====================================================================================================
Detection evaluation mAp=0.5895
====================================================================================================
SGG eval:     R @ 20: 0.2109;     R @ 50: 0.2451;     R @ 100: 0.2546;  for mode=sgcls, type=Recall(Main).
SGG eval:  ng-R @ 20: 0.3016;  ng-R @ 50: 0.4203;  ng-R @ 100: 0.4853;  for mode=sgcls, type=No Graph Constraint Recall(Main).
SGG eval:    zR @ 20: 0.0022;    zR @ 50: 0.0040;    zR @ 100: 0.0062;  for mode=sgcls, type=Zero Shot Recall.
SGG eval: ng-zR @ 20: 0.0050; ng-zR @ 50: 0.0160; ng-zR @ 100: 0.0360;  for mode=sgcls, type=No Graph Constraint Zero Shot Recall.
SGG eval:    mR @ 20: 0.1911;    mR @ 50: 0.2177;    mR @ 100: 0.2282;  for mode=sgcls, type=Mean Recall.
```

**Comment by hszhoushen** [comment]
Thanks for your rapid response. I found the reason why my previous training is not useful. I was using torch1.3 and loaded the pre-trained object detector at torch1.9 and save it therefore it can be used for torch1.3 as follows:

e.g.,
model = torch.load('model_name.pt')
torch.save(model, 'save_path', _use_new_zipfile_serialization=False)

However, after this process, the training process is not the same as your provided logs. I finally use torch1.9 and the pre-trained detector without re-saving, and now everything works properly.
